update_expected.py to parse artifacts and update graph break stats

wconstab opened the pull request:
Add a script that users can run to generate local copies of the .csvs for all the different model suites based on a particular CI run.  

Usage: 
1. Feed the sha of your commit to the script, after letting CI run on that sha (as a PR). 
2. The script should communicate with rocksDB and pull down data, generate csvs, and leave your git workspace in a 'dirty' state 
3. Verify the local changes to .csv files are expected, commit and push
4. Now, CI should pass on your PR (as the .csv should match the actual results)

TODO (cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire @ZainRizvi) hopefully i can convert the rocks query i'm using to a public API and delete the rocs api usage (and need for apikey) from this before landing.  If that's not easy or if i need to make a new query first, maybe i should land this as-is and at least people can use it if they get an apikey.  Also, any bad practices in how i parsed/mangled the filenames?  Would be nice to make the naming of artifacts more consistent with the job names so less mangling is needed. 

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96480
* #96346

wconstab committed:
WIP

[ghstack-poisoned]

wconstab committed:
Update on "WIP"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

wconstab committed:
Update on "WIP"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

wconstab committed:
Update on "update_expected.py to parse artifacts and update graph break stats"




[ghstack-poisoned]

wconstab committed:
Update on "update_expected.py to parse artifacts and update graph break stats"




[ghstack-poisoned]

ezyang commented:
I can't really review the rocks usage but the rest LGTM

wconstab committed:
Update on "update_expected.py to parse artifacts and update graph break stats"

TODO (cc ZainRizvi) hopefully i can convert the rocks query i'm using to a public API and delete the rocs api usage (and need for apikey) from this before landing.  If that's not easy or if i need to make a new query first, maybe i should land this as-is and at least people can use it if they get an apikey.  Also, any bad practices in how i parsed/mangled the filenames?  Would be nice to make the naming of artifacts more consistent with the job names so less mangling is needed. 




[ghstack-poisoned]

ZainRizvi commented on benchmarks/dynamo/ci_expected_accuracy/update_expected.py:
```diff
@@ -0,0 +1,140 @@
+def get_artifacts_urls(results, suites):
+    urls = {}
+    for r in results:
+        if "inductor" == r["workflowName"] and f"test" in r["jobName"]:
```
nit (linter would complain if run on this)

```suggestion
        if "inductor" == r["workflowName"] and "test" in r["jobName"]:
```

ZainRizvi commented on benchmarks/dynamo/ci_expected_accuracy/update_expected.py:
```diff
@@ -0,0 +1,140 @@
+
+
+def parse_test_str(test_str):
+    return (part.strip() for part in test_str[6:].strip(")k").split(","))
```
why strip the `k`? Typo?

wconstab commented on benchmarks/dynamo/check_graph_breaks.py:
```diff
@@ -57,8 +57,10 @@ def check_graph_breaks(actual_csv, expected_csv, expected_filename):
+        from pytorch/pytorch root, run
+        `python benchmarks/dynamo/ci_expected_accuracy/update_expected.py <sha>`
+        where <sha> points to a commit that has run inductor benchmark CI jobs (e.g. inductor_huggingface...)
+        and then include the resulting local changes to expected graph break CSVs to your commit.
```
yea, ill rephrase that way.

wconstab committed:
Update on "update_expected.py to parse artifacts and update graph break stats"

TODO (cc ZainRizvi) hopefully i can convert the rocks query i'm using to a public API and delete the rocs api usage (and need for apikey) from this before landing.  If that's not easy or if i need to make a new query first, maybe i should land this as-is and at least people can use it if they get an apikey.  Also, any bad practices in how i parsed/mangled the filenames?  Would be nice to make the naming of artifacts more consistent with the job names so less mangling is needed. 




[ghstack-poisoned]

ZainRizvi commented on benchmarks/dynamo/ci_expected_accuracy/update_expected.py:
```diff
@@ -0,0 +1,140 @@
+
+    params = {
+        "parameters": [
+            {
+                "name": "sha",
+                "type": "string",
+                "value": sha,
+            },
+            {"name": "repo", "type": "string", "value": repo},
```
nit: inconsistent formatting

ZainRizvi commented on benchmarks/dynamo/ci_expected_accuracy/update_expected.py:
```diff
@@ -0,0 +1,140 @@
+            suite, shard_id, num_shards, machine = parse_test_str(test_str)
+            workflowId = r["workflowId"]
+            id = r["id"]
+            runattempt = 1  # ? guessing here
```
You should be able to get the runattempt from rockset as well (IIRC, it's another column on the jobs/workflows table)

Not all jobs even have an attempt #1. For example, the first run attempt on PRs submitted by external branches will be `runattempt=2`

ZainRizvi commented on benchmarks/dynamo/ci_expected_accuracy/update_expected.py:
```diff
@@ -0,0 +1,140 @@
+import pandas as pd
+import requests
+
+
```
nit: I'd suggest adding a link to the artifacts query in rockset or something. The goal is to let a future dev find the exact query that this is pointing to, since it's really hard to do that reverse search with this public url

ZainRizvi commented on benchmarks/dynamo/ci_expected_accuracy/update_expected.py:
```diff
@@ -0,0 +1,140 @@
+        "inductor_huggingface",
+        "inductor_timm",
+        "inductor_torchbench",
+        # "inductor_huggingface_dynamic",
+        # "inductor_timm_dynamic",
+        # "inductor_torchbench_dynamic",
```
nit: delete if not needed

ZainRizvi commented on benchmarks/dynamo/ci_expected_accuracy/update_expected.py:
```diff
@@ -0,0 +1,140 @@
+        # "inductor_torchbench_dynamic",
+    }
+
+    # TODO check path and warn to run from pytorch root
```
nit: do you want to implement this before merging?

wconstab committed:
Update on "update_expected.py to parse artifacts and update graph break stats"

TODO (cc ZainRizvi) hopefully i can convert the rocks query i'm using to a public API and delete the rocs api usage (and need for apikey) from this before landing.  If that's not easy or if i need to make a new query first, maybe i should land this as-is and at least people can use it if they get an apikey.  Also, any bad practices in how i parsed/mangled the filenames?  Would be nice to make the naming of artifacts more consistent with the job names so less mangling is needed. 




[ghstack-poisoned]

wconstab committed:
Update on "update_expected.py to parse artifacts and update graph break stats"

TODO (cc ZainRizvi) hopefully i can convert the rocks query i'm using to a public API and delete the rocs api usage (and need for apikey) from this before landing.  If that's not easy or if i need to make a new query first, maybe i should land this as-is and at least people can use it if they get an apikey.  Also, any bad practices in how i parsed/mangled the filenames?  Would be nice to make the naming of artifacts more consistent with the job names so less mangling is needed. 




[ghstack-poisoned]

wconstab commented:
@pytorchbot merge

wconstab commented:
@pytorchbot merge
